sdk/docs/core/az_hfsm.md

```diff
@@ -0,0 +1,270 @@
+6. Transitions in `Enter` / `Exit` handlers are **not** supported.
+
+## Events
+
+Events are dynamically typed objects, optionally containing additional data required by the state machine.
+
+### Event Types
+
+The following public-API events are always supported. Additionally, HFSM-implementation specific events can be defined.
+
+```C
+enum az_event_type_generic
+{
+  /// Entry event: must not set or use the data field, must be handled by each state.
+  AZ_HFSM_EVENT_ENTRY = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 1),
+
+  /// Exit event: must not set or use the data field, must be handled by each state.
+  AZ_HFSM_EVENT_EXIT = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 2),
+
+  /// Generic error event: must use a data field containing a structure derived from
+  /// #az_hfsm_error_data
+  AZ_HFSM_EVENT_ERROR = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 3),
+
+  /**
+   * @brief Generic timeout event: if multiple timers are necessary it's recommended to create
+   * separate timeout events.
+   *
+   */
+  AZ_HFSM_EVENT_TIMEOUT = _az_HFSM_MAKE_EVENT(_az_FACILITY_HFSM, 4),
+};
+```
+
+### Event Data
+
+The HFSM framework defines the event-data corresponding to the `AZ_HFSM_EVENT_ERROR` only. Enter and Exit events do not have attached event-data (i.e. the parameter is set to NULL). Additionally, HFSM-implementation specific events can be defined.
+
+### Event Object Lifetime
+
+Events are always passed by value (on the stack). It is recommended that event data are modeled as C structs that avoid double-buffering of large data items (i.e. by relying on `az_span`).
+
+The application is responsible for maintaining run-to-completion semantics while the HFSM handler is running: any event-data must be available and remain unchanged throughout handler execution.
+
+## State Machine Composition
+
+While the hierarchical aspect already substantially decreases the number of states required, two different state composition patterns are also supported: bi-directional pipeline of HFSMs and orthogonal region.
+
+### Pipeline
+
+The pipeline is described in the [az_event_pipeline.md](az_event_pipeline.md) document. Within the SDK, the [Pipeline pattern](https://en.wikipedia.org/wiki/Pipeline_%28software%29) allows bi-directional message (event) passing within a chain of HFSM objects.
+
+### Orthogonal Region
+
+We implement the [Orthogonal region pattern](https://en.wikipedia.org/wiki/UML_state_machine#Orthogonal_regions) within the `az_event_policy_collection`. This works by multicasting the events to a subset of all the HFSMs in the collection based on an event-filter.
```

Comment:
```suggestion
We implement the [Orthogonal region pattern](https://wikipedia.org/wiki/UML_state_machine#Orthogonal_regions) within the `az_event_policy_collection`. This works by multicasting the events to a subset of all the HFSMs in the collection based on an event-filter.
```

Reply:
I'll leave it `en.`. I tried another language and because it isn't translated, it links to an empty page...